modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridLocalConfigManager.java

```diff
@@ -0,0 +1,333 @@
+                            cacheProcessor.splitter().split(storedCacheData.config());
+
+                        storedCacheData.config(splitCfg.get1());
+                        storedCacheData.cacheConfigurationEnrichment(splitCfg.get2());
+
+                        // Overwrite with new format.
+                        saveCacheConfiguration(storedCacheData, true);
+                    }
+
+                    String cacheName = storedCacheData.config().getName();
+
+                    CacheType type = ctx.cache().cacheType(cacheName);
+
+                    if (!caches.containsKey(cacheName))
+                        // No static cache - add the configuration.
+                        addStoredCache(caches, storedCacheData, cacheName, type, true, false);
+                    else {
+                        // A static cache with the same name already exists.
+                        CacheConfiguration cfg = caches.get(cacheName).cacheData().config();
+                        CacheConfiguration cfgFromStore = storedCacheData.config();
+
+                        validateCacheConfigurationOnRestore(cfg, cfgFromStore);
+
+                        addStoredCache(caches, storedCacheData, cacheName, type, true,
+                            cacheProcessor.keepStaticCacheConfiguration());
+
+                        if (!cacheProcessor.keepStaticCacheConfiguration() && type == CacheType.USER)
+                            skippedConfigs.add(cacheName);
+
+                    }
+                }
+
+                if (!F.isEmpty(skippedConfigs)) {
+                    U.warn(log, "Static configuration for the following caches will be ignored because a persistent " +
+                        "cache with the same name already exist (see " +
+                        "https://apacheignite.readme.io/docs/cache-configuration for more information): " +
+                        skippedConfigs);
+                }
+            }
+        }
+    }
+
+    /**
+     * Add stored cache data to caches storage.
+     *
+     * @param caches Cache storage.
+     * @param cacheData Cache data to add.
+     * @param cacheName Cache name.
+     * @param cacheType Cache type.
+     * @param isStaticallyConfigured Statically configured flag.
+     */
+    private void addStoredCache(
+        Map<String, CacheJoinNodeDiscoveryData.CacheInfo> caches,
+        StoredCacheData cacheData,
+        String cacheName,
+        CacheType cacheType,
+        boolean persistedBefore,
+        boolean isStaticallyConfigured
+    ) {
+        if (!caches.containsKey(cacheName)) {
+            if (!cacheType.userCache())
+                stopSeq.addLast(cacheName);
+            else
+                stopSeq.addFirst(cacheName);
+        }
+
+        caches.put(cacheName, new CacheJoinNodeDiscoveryData.CacheInfo(cacheData, cacheType, cacheData.sql(),
+            persistedBefore ? 1 : 0, isStaticallyConfigured));
```

Comment:
@glukos please avoid accepting magic code into the project. This constant meaning should be documented in code, e.g. introduced public static final fields with the particular bit meaning. Keeping it as-is become error-prone code